05a-run_ica.py

```diff
@@ -65,34 +65,34 @@ def run_ica(subject, tsss=config.mf_st_duration):
     epochs_for_ica = mne.Epochs(raw.copy().filter(l_freq=1., h_freq=None),
                                 events, config.event_id, config.tmin,
                                 config.tmax, proj=True, baseline=config.baseline,
-                                preload=False, decim=config.decim,
+                                preload=True, decim=config.decim,
                                 reject=config.reject)
 
     # run ICA on MEG and EEG
-    picks_meg = mne.pick_types(raw.info, meg=True, eeg=False,
+    picks_meg = mne.pick_types(epochs_for_ica.info, meg=True, eeg=False,
                                eog=False, stim=False, exclude='bads')
-    picks_eeg = mne.pick_types(raw.info, meg=False, eeg=True,
+    picks_eeg = mne.pick_types(epochs_for_ica.info, meg=False, eeg=True,
                                eog=False, stim=False, exclude='bads')
     all_picks = dict({'meg': picks_meg, 'eeg': picks_eeg})
 
     for ch_type in ['meg', 'eeg']:
         print('Running ICA for ' + ch_type)
 
         if ch_type == 'meg':
-
-            # XXX this does not work on epochs:
-            # meg_maxfilter_rank = epochs_for_ica.copy().pick_types(meg=True).estimate_rank()
-            n_components = 0.999
+            # XXX requires MNE 0.18
+            n_components = (mne.compute_rank(epochs_for_ica.copy()
+                            .pick_types(meg=True)))
 
         elif ch_type == 'eeg':
-
             n_components = 0.999
 
         ica = ICA(method='fastica', random_state=config.random_state,
                   n_components=n_components)
 
         picks = all_picks[ch_type]
-
+        
+        # XXX this breaks:
+        # *** TypeError: %i format: a number is required, not dict
         ica.fit(epochs_for_ica, picks=picks, decim=decim)
```

Comment:
@massich @agramfort 
I run into an error here that I wasn't able to fix. My hunch is that it came with using mne 0.18. 
Can you help?

  File "<ipython-input-11-906c8a461b22>", line 1, in <module>
    runfile('/Users/sophie/repos/mne-study-template/05a-run_ica.py', wdir='/Users/sophie/repos/mne-study-template')

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 704, in runfile
    execfile(filename, namespace)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 128, in <module>
    parallel(run_func(subject) for subject in config.subjects_list)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 128, in <genexpr>
    parallel(run_func(subject) for subject in config.subjects_list)

  File "/Users/sophie/repos/mne-study-template/05a-run_ica.py", line 93, in run_ica
    ica.fit(epochs_for_ica, picks=picks, decim=decim)

  File "</Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/externals/decorator.py:decorator-gen-365>", line 2, in fit

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/utils/_logging.py", line 89, in wrapper
    return function(*args, **kwargs)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 460, in fit
    self._fit_epochs(inst, picks, decim, verbose)

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 559, in _fit_epochs
    self._fit(data, self.max_pca_components, 'epochs')

  File "/Users/sophie/anaconda/envs/mne/lib/python3.6/site-packages/mne/preprocessing/ica.py", line 634, in _fit
    self.n_components)

TypeError: %i format: a number is required, not dict